Quantile filter

konikvranik opened the pull request:
## Description:

Documents quantile filter

**Related issue (if applicable):** fixes esphome/feature-requests#1543

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2900

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

mgm-pvranik committed:
quantile filter

mgm-pvranik committed:
formatting

oxan commented on components/sensor/index.rst:
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
+``quantile``
+************
+
+A `simple moving quantile <https://www.wikiwand.com/en/Quantile>`__
```
Please link to the regular Wikipedia site, instead of a third-party copy.

oxan commented on components/sensor/index.rst:
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
+
+Configuration variables:
+
+- **window_size** (*Optional*, int): The number of values over which to calculate the median
```
Copy/paste error, median -> quantile.

oxan commented on components/sensor/index.rst:
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
+
+- **window_size** (*Optional*, int): The number of values over which to calculate the median
+  when pushing out a value. This number should
+  be odd if you want an actual received value pushed out.
```
Copy/paste error, this doesn't apply to your filter.

oxan commented on components/sensor/index.rst:
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
+  be odd if you want an actual received value pushed out.
+  Defaults to ``5``.
+- **send_every** (*Optional*, int): How often a sensor value should be pushed out. For
+  example, in above configuration the median is calculated after every 4th
```
Copy/paste error, median -> quantile.

mgm-pvranik committed:
- link to original wiki page
- fixed copy paste errors (madian -> quantile)
- quantile param is optional with default.

oxan commented on components/sensor/index.rst:
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
+  published. With this parameter you can specify when the very first value is to be sent.
+  Must be smaller than or equal to ``send_every``
+  Defaults to ``1``.
+- **quantile** (**Optional**, float): value from 0 to 1 to determine which quantile to pick.
```
```suggestion
- **quantile** (*Optional*, float): value from 0 to 1 to determine which quantile to pick.
```

oxan committed:
Update components/sensor/index.rst
